diff -purN linux-raspberrypi-kernel_1.20170811-1.orig/arch/arm/boot/dts/overlays/i2c-rtc-overlay.dts linux-raspberrypi-kernel_1.20170811-1/arch/arm/boot/dts/overlays/i2c-rtc-overlay.dts
--- linux-raspberrypi-kernel_1.20170811-1.orig/arch/arm/boot/dts/overlays/i2c-rtc-overlay.dts	2017-08-08 22:44:22.000000000 +0900
+++ linux-raspberrypi-kernel_1.20170811-1/arch/arm/boot/dts/overlays/i2c-rtc-overlay.dts	2018-02-13 15:51:20.687138825 +0900
@@ -158,6 +158,21 @@
 		};
 	};
 
+	fragment@10 {
+		target = <&i2c_gpio>;
+		__dormant__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "okay";
+
+			rx8130: rx8130@32 {
+				compatible = "epson,rx8130";
+				reg = <0x32>;
+				status = "okay";
+			};
+		};
+	};
+
 	__overrides__ {
 		abx80x = <0>,"+0";
 		ds1307 = <0>,"+1";
@@ -169,6 +184,7 @@
 		pcf8523 = <0>,"+7";
 		pcf8563 = <0>,"+8";
 		m41t62 = <0>,"+9";
+		rx8130 = <0>,"+10";
 		trickle-diode-type = <&abx80x>,"abracon,tc-diode";
 		trickle-resistor-ohms = <&ds1339>,"trickle-resistor-ohms:0",
 					<&abx80x>,"abracon,tc-resistor";
